docs/ref/models/lookups.txt

```diff
@@ -94,6 +94,13 @@ following methods:
     ``compiler.compile(expression)`` should be used. The ``compiler.compile()``
     method will take care of calling vendor-specific methods of the expression.
 
+    ``kwargs`` may be added to update contents of ``self.extra`` which will
+    override the default ``template``- or ``function``-variables of the
+    ``Expression``-class. This is useful for 3rd-party database-backends that
+    may have different SQL-syntax. Simply override ``as_sql`` with your
+    backend-specific ``as_vendorname``-method and call the existing ``as_sql``-
+    method with the modified ``template``- or ``function``-parameter.
+
```

Comment:
This particular paragraph is tricky because it's referring to the Query Expression API and not to Expressions more generally (which Lookups are not). Here's my attempt at writing it:

"
``kwargs`` may be used to supply extra context to the ``as_sql`` method for that particular call. This can be useful for Expressions in particular, where ``template`` can be overridden and variables intended for interpolating the ``template`` string can be passed. 3rd-party backends may have different SQL-syntax so they can pass these differences through via ``kwargs``.

Example: ``self.as_sql(compiler, connection, template="%(function)s(%(arg)s)", function='Print', arg='Hello World')``
"